controllers/sriovnetworknodepolicy_controller.go

```diff
@@ -443,7 +443,10 @@ func (r *SriovNetworkNodePolicyReconciler) syncDaemonSet(cr *sriovnetworkv1.Srio
 		// DeepDerivative checks for changes only comparing non zero fields in the source struct.
 		// This skips default values added by the api server.
 		// References in https://github.com/kubernetes-sigs/kubebuilder/issues/592#issuecomment-625738183
-		if equality.Semantic.DeepDerivative(in.Spec, ds.Spec) {
+
+		// Note: with DeepDerivative, controller has issue not updating nodeAffinity correctly:
+		// https://bugzilla.redhat.com/show_bug.cgi?id=1914066, Chnange it to DeepEqual()
```

Comment:
Typo: "chnange".
I also wonder if we should delete the comment abovee

Reply:
I can delete previous comment if you prefer, just thought it might be useful for reference if later we hit the endless loop issue again.

Reply:
I don't have a strong opinion, I guess it's fine since the comment below explains it pretty clearly.

Reply:
fixed typo